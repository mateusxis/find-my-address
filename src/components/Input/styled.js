import styled from 'styled-components';
import InputMask from 'react-input-mask';

export const SInput = styled(InputMask)`
  display: block;
  margin: 0;
  padding: 0.8rem 1.6rem;
  color: inherit;
  width: 80%;
  font-family: inherit;
  font-size: 1.5rem;
  font-weight: inherit;
  line-height: 1.5;
  border: none;
  border-radius: 0.4rem;
  transition: box-shadow 300ms;
  color: #455a64;

  &::placeholder {
    color: #b0bec5;
  }

  &:hover,
  &:focus {
    outline: none;
    box-shadow: 0.2rem 0.8rem 1.6rem #5e35b1;
  }

  @media (min-width: 601px) {
    font-size: 1.8rem;
    line-height: 1.8;
    width: 90%;
  }

  @media (min-width: 961px) {
    font-size: 2.1rem;
    line-height: 2.1;
    width: 100%;
  }
`;
