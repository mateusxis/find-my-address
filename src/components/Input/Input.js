import { SInput } from './styled';

const Input = ({ placeholder, onChange, value, type }) => {
  return (
    <SInput
      mask="99999-999"
      maskPlaceholder="-"
      placeholder={placeholder}
      onChange={onChange}
      value={value}
      type={type}
    />
  );
};

export default Input;
