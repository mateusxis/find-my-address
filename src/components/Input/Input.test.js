import renderer from 'react-test-renderer';

import Input from './Input';
jest.mock('react-input-mask', () => jest.fn(() => <></>));

describe('Input', () => {
  it('should render Input', () => {
    const tree = renderer
      .create(<Input placeholder="Digit here" onChange={() => jest.fn()} value="Mathew" type="text" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
