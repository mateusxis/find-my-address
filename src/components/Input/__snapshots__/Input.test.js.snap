// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render Input 1`] = `null`;
